### DISCUSSION TITLE
ARGs aren't set when running DOCKERFILE

### DISCUSSION THREAD

**Discussion by AndrewWSHenry** [aquasecurity/kube-bench#1683] [Questions and Help]
hi,

I have not been able to build the kube-bench container using the instructions for "Running CIS benchmark in an EKS cluster" [https://github.com/aquasecurity/kube-bench/blob/main/docs/running.md#running-cis-benchmark-in-an-eks-cluster](url). I have run the docker build command on Ubuntu 24.04 and OpenSUSE LEAP 15.6 and run into the same problem of the KUBECTL_VERSION ARG not being set/available. causing the wget to error.

The instructions don't mention a need to set those values and a brief look seems to suggest that the makefile run by the dockerfile should be setting those values. Just wondering if anyone is aware of a reason why the KUBECTL_VERSION arg has no value? 

This is the relevant output.

> Dockerfile:14
> --------------------
>   12 |     # Add kubectl to run policies checks
>   13 |     ARG KUBECTL_VERSION TARGETARCH
>   14 | >>> RUN wget -O /usr/local/bin/kubectl "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl"
>   15 |     RUN wget -O kubectl.sha256 "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl.sha256"
>   16 |     # Verify kubectl sha256sum
> --------------------
> ERROR: failed to solve: process "/bin/sh -c wget -O /usr/local/bin/kubectl \"https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl\"" did not complete successfully: exit code: 8

**Comment by afdesk** [comment]
thanks for the report.
closed in favor #1684

**Comment by AndrewWSHenry** [comment]
Ran it again and realized the problem occurs earlier as the KubeBenchVersion ARG isn't set in the makefile. either.

> Step 10/33 : RUN make build && cp kube-bench /go/bin/kube-bench
> ---> Running in 7f9401965171
> fatal: not a git repository (or any of the parent directories): .git
> GOOS=linux CGO_ENABLED=0 go build -ldflags "-X github.com/aquasecurity/kube-bench/cmd.KubeBenchVersion=" -o kube-bench . 

Running the command from the console to get the kube-bench version returns this.

> ~/work/kube-bench> git describe --tags --abbrev
> v0.8.0-11-g7ea1d59

**Reply by afdesk** [reply]
@AndrewWSHenry thanks a lot!

oh, #1684 is still opened for this issue.

but I can't reproduce it. 
Could you share the log of building:
```sh
$ make build-docker IMAGE_NAME=k8s/kube-bench
docker build --build-arg BUILD_DATE=2024-10-04T08:20:59Z \
                     	--build-arg VCS_REF=c683e93 \
		     	--build-arg KUBEBENCH_VERSION=v0.8.0 \
	             	--build-arg KUBECTL_VERSION=1.31.0 \
			--build-arg TARGETARCH=arm64 \
                       	-t k8s/kube-bench .
...
```

> ~/work/kube-bench> git describe --tags --abbrev
> v0.8.0-11-g7ea1d59

makefile contains `0` for `--abbrev` flag:
```sh
$ git describe --tags --abbrev=0
```

**Reply by AndrewWSHenry** [reply]
Hi,

I ran the "make build-docker" command and the build was successful. Thank you for resolving this.

Andrew

```
Successfully built 3eb130cb1a1f
Successfully tagged k8s/kube-bench:latest
```
